app/javascript/components/workflows/student/exams/show/components/ExamCodeBox.tsx

```diff
@@ -118,15 +118,17 @@ export const Editor: React.FC<EditorProps> = (props) => {
   }, [instance, ...valueUpdate]);
 
   // EFFECT: refresh the instance if any item in refreshProps changes
-  useEffect(() => {
-    if (instance) {
-      if (cursor) {
-        instance.setCursor(cursor.line, cursor.ch, {
-          scroll: false,
-        });
+  useLayoutEffect(() => {
```

Comment:
I'm curious why `useLayoutEffect` here?  IIUC, `useLayoutEffect` is used when you definitely want something to run synchronously with React's rendering, but then we're using a `setTimeout`, which trivially ends that synchronization...?

Reply:
Yeah I was trying useLayoutEffect first, and then decided to just use setTimeout, but forgot to change the effect back.